plugins/input/udpserver/input_udp.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 iLogtail Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package udpserver
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/alibaba/ilogtail"
+	"github.com/alibaba/ilogtail/helper/decoder"
+	"github.com/alibaba/ilogtail/pkg/logger"
+	"github.com/alibaba/ilogtail/pkg/protocol"
+)
+
+type UDPServer struct {
+	Format        string
+	Address       string
+	MaxBufferSize int
+
+	context   ilogtail.Context
+	decoder   decoder.Decoder
+	addr      *net.UDPAddr
+	conn      *net.UDPConn
+	collector ilogtail.Collector
+}
+
+func (u *UDPServer) Init(context ilogtail.Context) (int, error) {
+	u.context = context
+	var err error
+	if u.decoder, err = decoder.GetDecoder(u.Format); err != nil {
+		return 0, err
+	}
+
+	host, portStr, err := net.SplitHostPort(u.Address)
+	if err != nil {
+		logger.Error(u.context.GetRuntimeContext(), "UDP_SERVER_ALARM", "illegal udp listening addr", u.Address, "err", err)
+		return 0, err
+	}
+	if host == "" {
+		host = "0.0.0.0"
+	}
+
+	ip, err := net.ResolveIPAddr("ip", host)
```

Comment:
The host variables should be given more description. Because The `net.ResolveIPAddr` API shows us that 
```
The address parameter can use a hostname, but this is not recommended because it will return at most one of the host name's IP addresses.
```